### ISSUE TITLE
reproduction of results

### ISSUE THREAD

**Issue by beifangdongdabao** [hewei98/NeuNcut#2]
Hi,

Your proposed method provides a very novel solution for spectral clustering.

However, I am currently unable to reproduce the results reported in your paper. 

Could you please let me know whether there are additional implementation details that were not disclosed, or whether the code on GitHub has not been fully updated?

**Comment by hewei98** [comment]
Thanks for your inquiry!

I'm not clear about the specific issues you encountered during your experiment reproduction, but there are a few aspects that may require attention. First, the clustering performance largely depends on the extracted features and affinity matrices. For example, the principle of MCR² (Yu et al., 2020) and SENet (Zhang et al., 2021) are adopted (please refer to their Git repositories) to learn representations and affinities and achieve optimal performance. Second, the box constraint of our proposed objective is satisfied by the Softmax (or Gumbel-Softmax) function, thus ensuring the loss is applied to the Softmax output (rather than the last linear layer output). Finally, the selection of the penalty weight (i.e., γ) should be done carefully, especially when the number of clusters (i.e., k) grows larger (e.g., on ImageNet dataset).&nbsp;This may because the growth of the Laplacian term and the orthogonal term in our loss function becomes imbalanced as the number of clusters increases. For more exmperimtal details, please refer to Table 1 of our paper.

Hope these tips help!
